backend/handlers/handlers.go

```diff
@@ -631,54 +631,78 @@ func (handlers *Handlers) getRatesHandler(_ *http.Request) (interface{}, error)
 func (handlers *Handlers) getConvertToFiatHandler(r *http.Request) (interface{}, error) {
 	from := r.URL.Query().Get("from")
 	to := r.URL.Query().Get("to")
-	amount := r.URL.Query().Get("amount")
-	amountAsFloat, err := strconv.ParseFloat(amount, 64)
-	if err != nil {
+	coin := r.URL.Query().Get("amount")
+	coinRat, valid := new(big.Rat).SetString(coin)
+	if !valid {
 		return map[string]interface{}{
 			"success": false,
 			"errMsg":  "invalid amount",
 		}, nil
 	}
+
 	rate := handlers.backend.RatesUpdater().LatestPrice()[from][to]
+	amount := new(big.Rat).Mul(coinRat, new(big.Rat).SetFloat64(rate))
+
 	return map[string]interface{}{
 		"success":    true,
-		"fiatAmount": strconv.FormatFloat(amountAsFloat*rate, 'f', 2, 64),
+		"fiatAmount": coinpkg.FormatAsCurrency(amount, to),
 	}, nil
 }
 
 func (handlers *Handlers) getConvertFromFiatHandler(r *http.Request) (interface{}, error) {
 	isFee := false
 	from := r.URL.Query().Get("from")
+	// TODO here we should check the `from` field, but there is not a list of valid fiats
```

Comment:
imo TODO's in code are very rarely useful and should only be added if the fix is coming very soon, if at all. i have seen too many permanent TODO's that stay forever or lose their meaning over time :sweat_smile: 

new tasks are best put into an issue tracker.

Reply:
ok! I added it to the issue I made to list the supported fiat (https://gitlab.com/shiftcrypto/godbb/-/issues/1337